File: binarysearch/Linked-List-Jumps/Linked-List-Jumps.cpp
// https://helloacm.com/recursive-algorithm-to-compute-the-linked-list-jumps/
// https://binarysearch.com/problems/Linked-List-Jumps
// EASY, RECURSION

/**
 * class LLNode {
 *     public:
 *         int val;
 *         LLNode *next;
 * };
 */
LLNode* solve(LLNode* node) {
    if (!node) return NULL;
    int jump = node->val;
    auto head = node;
    for (int i = 0; i < jump && (node); ++ i) {
        node = node->next;
    }
    head->next = solve(node);
    return head;
}

File: binarysearch/Linked-List-Jumps/Linked-List-Jumps.py
# https://helloacm.com/teaching-kids-programming-linked-list-jumps-via-recursion/
# https://binarysearch.com/problems/Linked-List-Jumps
# EASY, RECURSION

# class LLNode:
#     def __init__(self, val, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def solve(self, node):
        if not node:
            return node
        
        head = node
        for _ in range(head.val):
            head = head.next
            if not head:
                break
        
        node.next = self.solve(head)
        return node
